src/compile/mark/valueref.ts

```diff
@@ -80,10 +80,15 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   defaultRef: VgValueRef | 'zeroOrMin' | 'zeroOrMax'): VgValueRef {
   // TODO: datum support
 
+  // FIXME: walk through the logic here
+
   if (channelDef) {
     /* istanbul ignore else */
     if (isFieldDef(channelDef)) {
       if (hasDiscreteDomain(scale.type)) {
+        if (channelDef.bin) {
+          return fieldRef(channelDef, scaleName, {binSuffix: 'start'});
```

Comment:
Actually, if there should be a third case, maybe we should remove hasContinuousDomain / hasDiscreteDomain and replace them with `domainType` that returns one of the 3 string literals?  That will probably avoid misusing it.